Fix pm2-process restart error

peterMuriuki opened the pull request:
- use weareinteractive.pm2 role to install and configure pm2 processes
- Test using podman

peterMuriuki committed:
Re- adding vagrant as provisioner in test

It can be really stressful dealing with docker when you want to containerize a service that makes use of systemd

peterMuriuki committed:
Switch requirements to use podman

peterMuriuki committed:
Remove default scenario

peterMuriuki committed:
Uncomment nodejs system dependencies

peterMuriuki committed:
Save pm2 dump then kill

Somehow this way when the handler service start task runs it does not fail

peterMuriuki committed:
Refactor tasks to install node

peterMuriuki committed:
Add ubuntu1804 and ubuntu2004 scenarios

peterMuriuki committed:
Update yamlint

peterMuriuki committed:
Remove surplus files

peterMuriuki committed:
Use podman in place of docker in github actions

peterMuriuki commented:
created issue to look at idempotence and linting here: https://github.com/onaio/ansible-express/issues/13

moshthepitt commented:
what is causing the test to fail?

peterMuriuki committed:
Revert "Remove surplus files"

This reverts commit cbb776dd76d2cb8d849433f1f7ae0df6bf492fbc.

peterMuriuki committed:
Remove surplus files

peterMuriuki committed:
Install dependencies

moshthepitt commented on tasks/configure.yml:
```diff
@@ -41,77 +41,19 @@
-  ignore_errors: true
-  tags:
-    - molecule-idempotence-notest  # noqa 301
+- name: pm2 configure
```
Instead of doing this here, can we do it in the `meta`?  Here is are examples: [1](https://github.com/onaio/ansible-sqitch-migrations/blob/master/meta/main.yml#L19), [2](https://github.com/onaio/ansible-onadata/blob/master/meta/main.yml#L19).

peterMuriuki replied:
some of the envs that should be passed are outputs of havin run some tasks for example. pm2 start command needs to be passed the path where the index file is, the index file is an output of running build.
The problem is that I understand dependencies will run before the role actually executes.

moshthepitt replied:
cool, makes sense

peterMuriuki committed:
Trye out drone

Forced due to reason that ll containers running on github actions run on systemd-unsupported docker containers

peterMuriuki committed:
Target specific process when deploying

Make it possible to deploy several apps to the same host

ciremusyoka committed:
WIP don't use weareinteractive and change pm2 start command

ciremusyoka committed:
Ensure pm2 service is started

ukanga reviewed (commented):
Quite a number of places where code is commented out, could we either remove it or make relevant parts part of the code?

Are we using both Github actions and drone or only one of them is going to be relevant?
